zeroconf/_exceptions.py

```diff
@@ -22,28 +22,38 @@
 
 
 class Error(Exception):
-    pass
+    """Base class for all zeroconf exceptions."""
 
 
 class IncomingDecodeError(Error):
-    pass
+    """Exception when there is invalid data in an incoming packet."""
 
 
 class NonUniqueNameException(Error):
-    pass
+    """Exception when the name is already registered."""
 
 
 class NamePartTooLongException(Error):
-    pass
+    """Exception when the name is too long."""
 
 
 class AbstractMethodException(Error):
-    pass
+    """Exception when a required method is not implemented."""
 
 
 class BadTypeInNameException(Error):
-    pass
+    """Exception when the type in a name is invalid."""
 
 
 class ServiceNameAlreadyRegistered(Error):
-    pass
+    """Exception when a service name is already registered."""
+
+
+class EventLoopBlocked(Error):
```

Comment:
Shouldn't the base class also include `concurrent.futures.TimeoutError`?

Reply:
That is an implementation detail we don't want to expose to the consumer of this library as it could change in the future.  Ideally consumers of the api can catch `Error` or more specifically this one.

Reply:
After this update should we be able to see more detail as to why the timeout happened rather than just knowing that a timeout happened?
Particularly, [`async_request`](https://github.com/jstasiak/python-zeroconf/blob/master/zeroconf/_services/info.py#L460) timed out in my case, but I have no idea which part of that function hung. I think it was [`async_wait_for_start`](https://github.com/jstasiak/python-zeroconf/blob/master/zeroconf/_core.py#L147)
but I don't know the code base well enough to debug from there.
Is it possible to add logging to this area?

Reply:
> And in THAT function, we have [this line](https://github.com/jstasiak/python-zeroconf/blob/master/zeroconf/_core.py?rgh-link-date=2021-11-18T17%3A53%3A13Z#L163) which is probably causing the hang up.

That shouldn't take more than a few ms to run at most.  Is it somehow on your system?

Reply:
It seems like its more likely that zeroconf hasn't started up or there are multiple instances running that are trampling over each other.  How are you starting the zeroconf instance?

Reply:
Is what on my system? I was just going off my pychromecast logs which points to `async_request` as the direct place of hanging. The indirect cause of the hanging is unknown though.

Reply:
> Is what on my system?

https://github.com/jstasiak/python-zeroconf/blob/master/zeroconf/_core.py?rgh-link-date=2021-11-18T17%3A53%3A13Z#L163

Reply:
No it is not.